packages/typespec-go/src/tcgcadapter/types.ts

```diff
@@ -352,16 +353,6 @@ export class typeAdapter {
         this.types.set(decimalKey, decimalType);
         return decimalType;
       }
-      case 'eTag': {
```

Comment:
Was `eTag` and the other removed types converted to `string`?

Reply:
same from: https://github.com/Azure/typespec-azure/pull/1237

Reply:
move to: https://github.com/Azure/autorest.go/pull/1411

Reply:
The motivation for the change makes sense. However, my concern here is that we no longer have a predefined union of type names, so we don't know the possible "sub-type" when writing the code and are left with magic strings.

If it's not possible to define a union for these sub-types, can we at least define some constants, or document the possible sub-types someplace? Maybe it's already there in some form and I just missed it?

Reply:
sure. created an issue: https://github.com/Azure/typespec-azure/issues/1260.

Reply:
So the ask here is to document the `crossLanguageDefinitionId`s of those azure core types?
I do not think we should include a union in TCGC to enable that - the initial motivation of moving them out from the built-in type is trying not to make TCGC strongly bond with azure when we do not have to.
I think we could put those crossLanguageDefinitionId in a document for azure types.

Reply:
correct. we could put the doc here: https://azure.github.io/typespec-azure/docs/howtos/DataPlane%20Generation%20-%20DPG/07tcgcTypes

Reply:
Can you please add a link to this from the tcgc code so it's easy to find? Who will own maintaining it over time?

Reply:
after the init version, i could help to keep it up-to-date. the scalar types that need special handling should be limited.